docs/malware-scanner/README.md

```diff
@@ -1 +1,44 @@
-## *Currently empty, check back later or contribute yourself*
\ No newline at end of file
+# Malware scanner and detector
+
+---
+
+# Dynamic analysis 
+A slow but reliable detector looking for jar capabilities, potential malware sign
+- run static analysis to filter known threats
+- downloading stuff from the internet (potentially jars, but can be genuine)
+- zip/jar manipulation.
+- able to execute stuff (Runtime.exec())
+- de-obfuscation, static string rebuilding from assets, byte arrays or StringBuilders
+- TODO 
+
+---
+
+# Static analysis from up-to-date signature database
+This should be used by modpack checkers/launchers, this should be fast, detect known threats, quickly filter malwares from huge directories (modpacks, whole disks)
+
+## Matching rules
+Standard way for representing JVM malware signatures.  
+Rules should be similar to Yara rules (just for JVM application) This means
+- [ ] Binary sequence matches (mostly for asset matching)
+- [ ] Filtered Java ASM instruction sequences (for example only check for method invocations, but all of them)  
+Ability to match only in selected functions could be useful
+- [ ] filename matches (with regex)
+- [ ] function name/id matches  
+- [ ] Match conditions like "at least 10" or "$a or $b and $c"
+
+### Signature database hosting, fetching  
+Signature database should be hosted somewhere for allowing the algorithm to quickly "learn" new malwares.  
+Executing anything from database is dangerous, see [Runtime Downloading](/docs/runtime-downloading/README.md).  
+Any downloaded *data* should be signed to prevent any misuse.
+
+
+## Detection algorithm  
+Should be fast **and** reliable. Reference implementation should be in JVM language.
+- [ ] fast matching, like [KMP](https://en.wikipedia.org/wiki/Knuth%E2%80%93Morris%E2%80%93Pratt_algorithm) O(m+n)
+- [ ] obfuscation resilent: ignore class hashes when loading file, open any .class what may be executed  
+- [ ] low-footpring, threading for speed
+- [ ] standalone/CLI interface
+- [ ] standalone/GUI interface + installer
```

Comment:
GUI/CLI actions I think should not be under `Detection algorithm` as a section

Perhaps make a new section for _"user consumables"_ or something specific to what end-users interact with, vs algorithm details and library usage.